File: Swift/sort/bubble/1/main.swift
//
//  main.swift
//  S
//
//  Created by Fay on 2017/7/21.
//  Copyright © 2017年 Fay. All rights reserved.
//

import Foundation

func bubbleSort(_ array: Array<Int>) -> Array<Int> {
    var arr = array
    var index = 0
    repeat {
        for i in 1 ..< arr.count {
            if (arr[i - 1] > arr[i]) {
                swap(&arr[i], &arr[i - 1])
            }
            index = i
        }
    } while index == arr.count
    return arr
}

var array = [2, 1, 3, 5, 4]
print(bubbleSort(array))

File: PHP/sort/bubble/1/main.php
<?php
/**
 * Copyright (c) 2018 faylib.top
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

function bubble_sort(&$array) {
    for ($i = 0; $i < count($array) - 1; $i++)
        for ($j = 0; $j < count($array) - 1 - $i; $j++)
            if ($array[$j] > $array[$j + 1])
                swap($array[$j], $array[$j + 1]);
    return $array;
}

function swap(&$i, &$j) {
    $temp = $i;
    $i = $j;
    $j = $temp;
}

$array = array(2, 1, 3, 5, 4);
echo join(' ', bubble_sort($array))."\n";
